gateway/api.go

```diff
@@ -3176,8 +3176,13 @@ func ctxSetCheckLoopLimits(r *http.Request, b bool) {
 
 // Should we check Rate limits and Quotas?
 func ctxCheckLimits(r *http.Request) bool {
+	// If this is a self loop, do not need to check the limits and quotas.
+	if ctxSelfLooping(r) {
+		return false
+	}
+
 	// If looping disabled, allow all
-	if !ctxLoopingEnabled(r) {
+	if ctxLoopingEnabled(r) {
```

Comment:
`ctxCheckLimits` is used to decide whether to check quota/rate limit checks, when looping is disabled (for direct proxy), it should check for rate limits. 
I don't think this change is required. 
raised https://github.com/TykTechnologies/tyk/pull/6780 without this change and tests are passing. 

I think we need to extend the test suite to check the case where separate auth tokens are used when request it looped to different API